cmd/keymasterd/certgen.go

```diff
@@ -157,6 +157,39 @@ func (state *RuntimeState) certGenHandler(w http.ResponseWriter, r *http.Request
 	}
 }
 
+// returns 3 values, if the key is valid, if the key is not valid, the text reason why and and error if it was an internal error
+
+func getValidSSHPublicKey(userPubKey string) (ssh.PublicKey, string, error) {
```

Comment:
Since the string is really an error message, why not make this an `error` type instead?

Reply:
because is then hard to tell if it is a validation error OR a user error. I am all ears on how to make this more consumable

Reply:
I'm not suggesting merging the string and the error returns. I'm suggesting to have both of these error-like returns be of type `error`.